ledger/test/test_compact_serializer.py

```diff
@@ -0,0 +1,228 @@
+
+def testSerializeSimpleJson():
+    serializer = CompactSerializer(fields)
+    assert b"v1|v2|3|4.0|True" == \
+           serializer.serialize(
+               {"f1": "v1", "f2": "v2", "f3": 3, "f4": 4.0, "f5": True}
+           )
+
+
+def testDeserializeSimpleJson():
+    serializer = CompactSerializer(fields)
+
+    assert {"f1": "v1", "f2": "v2", "f3": 3, "f4": 4.0, "f5": True} == \
+            serializer.deserialize(b"v1|v2|3|4.0|True")
+
+
+def testSerializeDeserializeSimpleJson():
+    serializer = CompactSerializer(fields)
+
+    json = {"f1": "v1", "f2": "v2", "f3": 3, "f4": 4.0, "f5": True}
+    serialized = serializer.serialize(json)
+    deserialized = serializer.deserialize(serialized)
+
+    assert json == deserialized
+
+
+def testSerializeToBytes():
+    serializer = CompactSerializer(fields)
+    assert b"v1|v2|3|4.0|True" == \
+           serializer.serialize(
+               {"f1": "v1", "f2": "v2", "f3": 3, "f4": 4.0, "f5": True},
+               toBytes=True
+           )
+
+def testSerializeToString():
+    serializer = CompactSerializer(fields)
+    assert "v1|v2|3|4.0|True" == \
+           serializer.serialize(
+               {"f1": "v1", "f2": "v2", "f3": 3, "f4": 4.0, "f5": True},
+               toBytes=False
+           )
+
+def testDeserializeFromBytes():
+    serializer = CompactSerializer(fields)
+
+    assert {"f1": "v1", "f2": "v2", "f3": 3, "f4": 4.0, "f5": True} == \
+            serializer.deserialize(b"v1|v2|3|4.0|True")
+
+
+def testSerializeToFields():
+    serializer = CompactSerializer(fields)
+
+    newFields = OrderedDict([
+        ("ff1", (str, str)),
+        ("ff2", (str, str)),
+        ("ff3", (str, int))
+    ])
+    assert b"v1|v2|3" == \
+           serializer.serialize(
+               {"ff1": "v1", "ff2": "v2", "ff3": 3},
+               fields=newFields
+           )
+
+def testDeserializeFromString():
+    serializer = CompactSerializer(fields)
+
+    assert {"f1": "v1", "f2": "v2", "f3": 3, "f4": 4.0, "f5": True} == \
+            serializer.deserialize("v1|v2|3|4.0|True")
+
+def testDeserializForFields():
+    serializer = CompactSerializer(fields)
+
+    newFields = OrderedDict([
+        ("ff1", (str, str)),
+        ("ff2", (str, str)),
+        ("ff3", (str, int))
+    ])
+    assert {"ff1": "v1", "ff2": "v2", "ff3": 3} == \
+            serializer.deserialize("v1|v2|3", fields=newFields)
+
+
+def testSerializeLessFields():
+    serializer = CompactSerializer(fields)
+
+    assert b"|v1|2||" == serializer.serialize({"f2": "v1", "f3": 2})
+    assert b"v1||||" == serializer.serialize({"f1": "v1"})
+    assert b"||||" == serializer.serialize({})
+    assert b"||||3" == serializer.serialize({"f5": 3})
+    assert b"v1||||3" == serializer.serialize({"f1": "v1", "f5": 3})
+
+
+def testDeserializeLessFields():
+    serializer = CompactSerializer(fields)
+
+    assert {"f1": None, "f2": "v1", "f3": 2, "f4": None, "f5": None} == serializer.deserialize(b"|v1|2||")
```

Comment:
Put a TODO comment about this. We need to fix it moving to a txn versioning and using another serialization (protobuf?)